src/interactive-window/interactiveWindow.ts

```diff
@@ -110,33 +107,85 @@ export class InteractiveWindow implements IInteractiveWindowLoadable {
     } = {};
     private pendingNotebookScrolls: NotebookRange[] = [];
 
+    private _notebookEditor!: NotebookEditor;
+    public get notebookEditor(): NotebookEditor {
+        return this._notebookEditor;
+    }
+
+    private _notebookUri: Uri;
+    public get notebookUri(): Uri {
+        return this._notebookUri;
+    }
+
+    private readonly documentManager: IDocumentManager;
+    private readonly fs: IFileSystem;
+    private readonly configuration: IConfigurationService;
+    private readonly jupyterExporter: INotebookExporter;
+    private readonly workspaceService: IWorkspaceService;
+    private readonly exportDialog: IExportDialog;
+    private readonly notebookControllerSelection: IControllerSelection;
+    private readonly interactiveWindowDebugger: IInteractiveWindowDebugger | undefined;
+    private readonly errorHandler: IDataScienceErrorHandler;
+    private readonly codeGeneratorFactory: ICodeGeneratorFactory;
+    private readonly storageFactory: IGeneratedCodeStorageFactory;
+    private readonly debuggingManager: IInteractiveWindowDebuggingManager;
+    private readonly isWebExtension: boolean;
+    private readonly commandManager: ICommandManager;
     constructor(
-        private readonly documentManager: IDocumentManager,
-        private readonly fs: IFileSystem,
-        private readonly configuration: IConfigurationService,
-        private readonly commandManager: ICommandManager,
-        private readonly jupyterExporter: INotebookExporter,
-        private readonly workspaceService: IWorkspaceService,
-        private _owner: Resource,
-        private mode: InteractiveWindowMode,
-        private readonly exportDialog: IExportDialog,
-        private readonly notebookControllerSelection: IControllerSelection,
         private readonly serviceContainer: IServiceContainer,
```

Comment:
I'm not a fan of passing the serviceContainer to constructors since it makes the class more coupled with the specific IoC container, but I suppose it's ok since it was already being passed in anyway

Reply:
Agree, I'm not a fan either. My motivation is avoiding doing `serviceContainer.get<>` in multiple places where we call `InteractiveWindow#ctor`. 

@DonJayamanne pointed out that it's explicit to us how many services external `InteractiveWindow` import but moving `serviceContainer.get<>` into `ctor` kinds of hiding it. We might want to revisit this along with other challenges I ran into (will leave a comment about this soon)